dali/base/dasds.cpp

```diff
@@ -289,6 +290,119 @@ class CLCWriteLockBlock : public CLCLockBlock
 #define CHECKEDDALIWRITELOCKBLOCK(l,timeout)  WriteLockBlock glue(block,__LINE__)(l)
 #endif
 
+class CLockFilters : public CInterface
+{
+    DALockTypeFilter lockType;
+    int durationLow;
+    int durationHigh;
+    StringAttr endpoint;
+
+    bool beginWith(const char*& ln, const char* pat)
+    {
+        size32_t sz = strlen(pat);
+        if (memicmp(ln, pat, sz))
+            return false;
+
+        ln += sz;
+        return true;
+    }
+public:
+    CLockFilters()
+    {
+        lockType = DALTFAll;
+        endpoint.set("*");
+        durationLow = -1;
+        durationHigh = -1;
+    };
+    void readFilters(const char *filterStr)
+    {
+        if (!filterStr || !*filterStr)
+            return;
+        StringArray filterStringArray;
+        char sep[] = { DALockFilterSeparator, '\0' };
+        filterStringArray.appendList(filterStr, sep);
+
+        unsigned filterFieldsToRead = filterStringArray.length();
+        ForEachItemIn(i,filterStringArray)
+        {
+            const char* filterTypeStr = filterStringArray.item(i);
+            if (!filterTypeStr || !*filterTypeStr || !isdigit(*filterTypeStr))
+                continue;
+            unsigned filterSize = 2;
+            DALockFilterType filterType = (DALockFilterType) atoi(filterTypeStr);
```

Comment:
Should parse human readable form of options, that could be passed on CLI.
e.g. : daliadmin . filteredlocks locktype=all|endpoint=..